WMF Framework/Remote Notifications/NotificationServiceHelper.swift

```diff
@@ -0,0 +1,64 @@
+              return false
+          }
+        
+        for notification in notifications {
+            if (notification.titleFull != firstTitle) ||
+                (notification.namespace != firstNamespace) {
+                return false
+            }
+        }
+        
+        return true
+    }
+    
+    public static func determineNotificationsToDisplayAndCache(fetchedNotifications: Set<RemoteNotificationsAPIController.NotificationsResult.Notification>, cachedNotifications: Set<RemoteNotificationsAPIController.NotificationsResult.Notification>) -> (notificationsToDisplay: Set<RemoteNotificationsAPIController.NotificationsResult.Notification>, notificationsToCache: Set<RemoteNotificationsAPIController.NotificationsResult.Notification>) {
+
+        //Prune persisted keys of any > 1 day ago so the cache doesn't grow too big
+        let recentCachedNotifications = cachedNotifications.filter { $0.isNewerThan(timeAgo: TimeInterval.oneDay) }
+        
+        //Prune fetched notifications > 10mins ago since a server delay should be no longer than that.
+        let recentFetchedNotifications = fetchedNotifications.filter { $0.isNewerThan(timeAgo: TimeInterval.tenMinutes) }
+        
+        //Only consider new notifications that don't exist in cache for display
+        let notificationsToDisplay = recentFetchedNotifications.subtracting(recentCachedNotifications)
+        
+        //New cache should keep track of recently cached notifications + new notifications to display
+        let notificationsToCache = notificationsToDisplay.union(recentCachedNotifications)
+        
+        return (notificationsToDisplay, notificationsToCache)
+    }
+    
+    public static func talkPageContent(for notifications: Set<RemoteNotificationsAPIController.NotificationsResult.Notification>) -> (subtitle: String, body: String)? {
+        
+        guard NotificationServiceHelper.allNotificationsAreForSameTalkPage(notifications: notifications),
+              let talkPageTitle = notifications.first?.titleFull else {
+            return nil
+        }
+        
+        let subtitle: String
+        let body: String
+        
+        let messageText = String.localizedStringWithFormat(WMFLocalizedString("notifications-push-talk-messages-format", value: "{{PLURAL:%1$d|message|messages}}", comment: "Plural messages text to be inserted into push notification content - %1$d is replaced with the number of talk page messages."), notifications.count)
+        subtitle = String.localizedStringWithFormat(WMFLocalizedString("notifications-push-talk-title-format", value: "New %1$@", comment: "Title text for a push notification that represents talk page messages - %1$@ is replaced with \"Messages\" text (can be plural or singular)."), messageText)
+        
+        if let first = notifications.first,
+           notifications.count == 1,
+           let pushContentText = first.pushContentText {
```

Comment:
```suggestion
        if notifications.count == 1,
           let pushContentText = notifications.first?.pushContentText {
```

Reply:
@mcleinman Thanks! Good suggestions - I added these, let me know if I missed anything (or feel free to push a commit).